src/copy.rs

```diff
@@ -0,0 +1,120 @@
+// Copyright (c) 2023-present, PingCAP, Inc. Licensed under Apache-2.0.
+
+use std::fs::{copy, create_dir_all};
+use std::os::unix::fs::symlink;
+use std::path::Path;
+use std::sync::Arc;
+
+use crate::config::Config;
+use crate::env::{FileSystem, Permission};
+use crate::file_pipe_log::{FileNameExt, FilePipeLogBuilder};
+use crate::pipe_log::{FileId, LogQueue};
+
+/// Make a copy from `source` to `target`. `source` should exists but `target`
+/// shouldn't.
+///
+/// *minimum* means *symlink* will be used if possbile, otherwise *copy* will be
+/// used instead. Generally all inactive log files will be symlinked, but the
+/// last active one will be copied.
+///
+/// After the copy is made both of 2 engines can be started and run at the same
+/// time.
+pub fn minimum_copy<F, P>(cfg: &Config, fs: Arc<F>, target: P) -> Result<(), String>
```

Comment:
I prefer it be an engine method, and hooked to engine-ctl, like `consistency_check`.

Reply:
A better name would be "fork".